text/0075-span-thread-id.md

```diff
@@ -0,0 +1,91 @@
+
+# Motivation
+
+Connecting profiling and performance products without thread information is currently resulting in a confusing and not very accurate user experience - for example, profiling always shows the entire span tree even though the "user view" is scoped to only a single thread. Similarly, we are attempting to show functions executed during the lifetime of a span in our performance product, but without thread information, we are left at guessing the best thread to show or havign to pick a default value (main) which is inaccurate at best. This RFC outlines the changes required to add the necessary span information that would enable us to improve that experience on both products.
+
+# Supporting Data
+
+Inaccurate and bad user experience in profiling and performance.
+
+Metrics:
+No usage metrics, but this was requested by customers.
+
+# Options Considered
+
+This section outlines what data we need to collect and which field to store it under followed by possible optimizations and benchmarks at the end.
+
+In order to accurately determine which spans were part of which thread and what code was being executed during the lifetime of a span, we need a way to accurately identify the two. An identifier that gives us sufficient accuracy is the thread identifier, a nice benefit of which is that it will remain unique across different processes which might be running our SDKs. Collecting thread id information also seems like standard practice in open telemetry (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-thread-attributes), so we would not be inventing anything new by adopting it - if anything, it brings us closer to otel which also marks thread.id and thread.name as recommended instead of required attributes.
+
+Marking thread id as a recommended attributes also gives us flexibility in terms of only collecting the data while there is an active profiler session - I currently do not have strong reasoning to suggest that we should be collecting it all the time, if you think it could be beneficial, please add a comment.
+
+The common data type for a thread identifier is uint64 (also the one we currently use in profiling) and one that we likely should be using.
+
+Suggestion:
+- collect thread.id on spans regardless if profiling is enabled or not
+- collect thread.id on spans only during an active profiling session
+
+
+Where do we store the data?
+There are two places where we could store thread information - either on the span data dict and using setData or by adding new fields to the span object. The former has the benefit of a prexisting API, we would only have to ensure the data is properly set while the later seems more involved and might have infrastructure implications. Since the data dict is public, we would probably want to ensure that users do not override the thread information or discard it if they do - it sounds like acceptable risk to me.
```

Comment:
For file I/O on the main thread we use `setData`. We agreed on the key `blocked_main_thread`, see https://github.com/getsentry/develop/blob/1c1b528a3442405dad51eedef6e69dbbec296d24/src/docs/sdk/performance/span-data-conventions.mdx?plain=1#L24

Reply:
All span data documented here: https://develop.sentry.dev/sdk/performance/span-data-conventions/